src/components/FileUploaderV2/FileUploaderV2.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import uid from '../../tools/uniqueId';
+import { ButtonTypes } from '../../prop-types/types';
+import FileUploadStatus from '../FileUploadStatus/FileUploadStatus';
+
+export class FileUploaderButtonV2 extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    labelText: PropTypes.string,
+    multiple: PropTypes.bool,
+    name: PropTypes.string,
+    onChange: PropTypes.func,
+    onClick: PropTypes.func,
+    role: PropTypes.string,
+    tabIndex: PropTypes.number,
+    buttonKind: ButtonTypes.buttonKind,
+    accept: PropTypes.arrayOf(PropTypes.string),
+  };
+  static defaultProps = {
+    tabIndex: 0,
+    labelText: 'Add file',
+    buttonKind: 'primary',
+    multiple: false,
+    onChange: () => {},
+    onClick: () => {},
+    accept: [],
+  };
+  uid = this.props.id || uid();
+
+  render() {
+    const {
+      className,
+      multiple,
+      role,
```

Comment:
do you mean we no longer want to destructure these props @asudoh ?

Reply:
@emyarod Thank you for asking - `role` and `tabIndex` seem to be set by `{...others}`, and thus we don't need to set those explicitly. Does it make sense? Don't hesitate to ping me otherwise!

Reply:
got it @asudoh , and just to make sure, is that what you meant in [this comment](https://github.com/IBM/carbon-components-react/pull/1105#discussion_r215134786) as well?

Reply:
@emyarod Yes - I’d say any props that are covered by `{…other}` don’t have to be set explicitly (except `propTypes`).